module/proto/peggy/v1/msgs.proto

```diff
@@ -110,114 +47,27 @@ message MsgSendToEth {
 
 message MsgSendToEthResponse {}
 
-// MsgRequestBatch
-// this is a message anyone can send that requests a batch of transactions to
-// send across the bridge be created for whatever block height this message is
-// included in. This acts as a coordination point, the handler for this message
-// looks at the AddToOutgoingPool tx's in the store and generates a batch, also
-// available in the store tied to this message. The validators then grab this
-// batch, sign it, submit the signatures with a MsgConfirmBatch before a relayer
-// can finally submit the batch
-// -------------
 message MsgRequestBatch {
-  string orchestrator = 1;
-  string denom        = 2;
+  string validator = 1;
+  string denom     = 2;
 }
 
 message MsgRequestBatchResponse {}
 
-// MsgConfirmBatch
-// When validators observe a MsgRequestBatch they form a batch by ordering
-// transactions currently in the txqueue in order of highest to lowest fee,
-// cutting off when the batch either reaches a hardcoded maximum size (to be
-// decided, probably around 100) or when transactions stop being profitable
-// (TODO determine this without nondeterminism) This message includes the batch
-// as well as an Ethereum signature over this batch by the validator
-// -------------
-message MsgConfirmBatch {
-  uint64 nonce          = 1;
-  string token_contract = 2;
-  string eth_signer     = 3;
-  string orchestrator   = 4;
-  string signature      = 5;
-}
-
-message MsgConfirmBatchResponse {}
-
-// MsgConfirmLogicCall
-// When validators observe a MsgRequestBatch they form a batch by ordering
-// transactions currently in the txqueue in order of highest to lowest fee,
-// cutting off when the batch either reaches a hardcoded maximum size (to be
-// decided, probably around 100) or when transactions stop being profitable
-// (TODO determine this without nondeterminism) This message includes the batch
-// as well as an Ethereum signature over this batch by the validator
-// -------------
-message MsgConfirmLogicCall {
-  string invalidation_id = 1;
-  uint64 invalidation_nonce = 2;
-  string eth_signer     = 3;
-  string orchestrator   = 4;
-  string signature      = 5;
-}
-
-message MsgConfirmLogicCallResponse {}
-
-// EthereumBridgeDepositClaim
-// When more than 66% of the active validator set has
-// claimed to have seen the deposit enter the ethereum blockchain coins are
-// issued to the Cosmos address in question
-// -------------
-message MsgDepositClaim {
-  uint64 event_nonce     = 1;
-  uint64 block_height    = 2;
-  string token_contract  = 3;
-  string amount          = 4 [
-    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Int", 
-    (gogoproto.nullable) = false
+message MsgSubmitClaim {
+  google.protobuf.Any claim = 1 [
+      (cosmos_proto.accepts_interface) = "EthereumClaim"
```

Comment:
Is the reason for using any here to allow other people to plug in their own claim types? If we want to allow users to only use the provided claim types we should use a oneof here. 

Its better to opinionated in a few things to reduce the complexity of such a system.

Reply:
Thats a great point! And sounds like we should use a OneOf